### PR TITLE
Change random seed interface

### PR DESCRIPTION
**Pull request by AldoGl**
## Proposed changes

I changed the random seed structure of the calibrator and of the samplers. This change was already present in #2 but I am reproducing it here to ease code review.

Now the seed given to the calibrator is used to initialise the seeds of the samplers in a random (yet reproducible) way, and it is also used to initialise a random generator hat gives different seeds to the model simulations.

I broke down the PR in different commits to facilitate review.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/master/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by AldoGl** [comment]
I don't understand why pylint gives these import errors, it works well on my machine, any idea?

File: black_it/samplers/base.py [hunk 1]
```diff
@@ -34,20 +35,40 @@ class BaseSampler(ABC):
     """
 
     def __init__(
-        self, batch_size: int, internal_seed: int = 0, max_duplication_passes: int = 5
+        self, batch_size: int, random_state: int = 0, max_duplication_passes: int = 5
```
Comment (marcofavoritobi) [inline black_it/samplers/base.py:38]: should be `random_state: Optional[int] = None`

Comment (marcofavoritobi) [inline black_it/samplers/base.py:38]: should be `random_state: Optional[int] = None`

Comment (marcofavoritobi) [inline black_it/samplers/base.py:38]: should be `random_state: Optional[int] = None`

File: tests/test_samplers/test_gaussian_process.py [hunk 2]
```diff
@@ -96,6 +100,7 @@ def test_gaussian_process_2d(
                 "The test with acquisition=expected_improvement is temporarily "
                 "skipped due to reproducibility issues."
             )
+        print(acquisition, optimize_restarts, new_params)
```
Comment (marcofavoritobi) [inline tests/test_samplers/test_gaussian_process.py:103]: to remove

**Pull request merged**